.ci/scripts/process_metric_data.py

```diff
@@ -36,27 +41,214 @@ def remove_unwanted_lines(csv_file):
+    elif "TB" in meas_str:
+        numeric_val = float(meas_str.split("TB")[0]) * 1000
+    elif "B" in meas_str:
+        numeric_val = float(meas_str.split("B")[0]) / 1000000000
+    else:
+        print(f"unexpected string value {meas-str}")
+        numeric_val = meas_str
+    return numeric_val
+
+
+def get_net_send_recv(net_str):
+    """Split into blocks sent and received."""
+    s, r = net_str.split(' / ')
+    s = conv_to_gb(s)
+    r = conv_to_gb(r)
+    return s, r
+
+
+def get_disk_read_write(block_str):
+    """Split into blocks read and written."""
+    r, w = block_str.split(' / ')
+    r = conv_to_gb(r)
+    w = conv_to_gb(w)
+    return r, w
+
+
+def format_out_row_misc(misc_row):
+    """
+       Return a formatted, comma separated row of docker miscellaneous data.
+
+       Parameters
+       ----------
+       misc_row : dictionary
+           row from the data collection of miscellaneous stats
+
+       Returns
+       -------
+       formatted, comma separated string
+
+       """
+
+    secs = misc_row['SECONDS']
+    disk = get_disk_gb(misc_row['disk_used'])
+    # swap = dr['swap_used'].split()[2]
+    threads = misc_row['total_threads']
+    # Todo fix this code so we get the last line stuff: right now it's blank
+    # only update last line if it has changed
+    global prior_log_line
+    if prior_log_line is not None and misc_row['last_line'] == prior_log_line:
+        last_line = ""
+    else:
+        last_line = misc_row['last_line']
+        prior_log_line = last_line
+    return f"{secs},{disk},{threads},{last_line}"
+
+
+def format_out_row_docker(stats_row):
+    """
+    Return a formatted, comma separated row of docker stats data.
+
+    Parameters
+    ----------
+    stats_row : dictionary
+        row from data collection of docker stats
 
+    Returns : string
+    -------
+        formatted, comma separated string
 
-def main(csv_file):
+    """
+
+    secs = stats_row['SECONDS']
+    name = stats_row['{{.Name}}']
+    pids = stats_row['{{.PIDs}}']
+    cpu = stats_row['{{.CPUPerc}}'].replace("%", "")
+    mem = get_mem_gb(stats_row['{{.MemUsage}}'])
+    mem_p = stats_row['{{.MemPerc}}'].replace("%", "")
+    net_s, net_r = get_net_send_recv(stats_row['{{.NetIO}}'])
+    disk_r, disk_w = get_disk_read_write(stats_row['{{.BlockIO}}'])
+    return f"{secs},{name},{pids},{cpu},{mem},{mem_p},{net_s},{net_r},{disk_r},{disk_w}"
+
+
+def make_lists(csv_file):
+    """
+    Makes a list of dictionaries for each row of the csv file captured during the docker run
+
+    Parameters
+    ----------
+    csv_file : file
+        file used to capture either docker stats or miscellaneous OS measurements
+
+    Returns
+    -------
+    csv_to_list : list
+        List that is now ready to be formatted
+
+    """
+    csv_to_list = []
+    with open(csv_file) as csv_handle:
+        reader = csv.DictReader(csv_handle)
+        for row in reader:
+            csv_to_list.append(row)
+    return csv_to_list
```

Comment:
Could condense this function a bit:

```python
with open(csv_file) as csv_handle:
    return [row for row in csv.DictReader(csv_handle)]
```

Reply:
Cool application of a list comprehension.   I'll look for opportunities to use.